tracer/src/Datadog.Trace/Logging/TracerFlare/TracerFlareManager.cs

```diff
@@ -0,0 +1,360 @@
+    {
+        if (Interlocked.Exchange(ref _subscription, new Subscription(RcmProductReceived, RcmProducts.TracerFlareInitiated, RcmProducts.TracerFlareRequested)) == null)
+        {
+            _discoveryService.SubscribeToChanges(HandleConfigUpdate);
+            // Don't need to set any capabilities for tracer flare
+            _subscriptionManager.SubscribeToChanges(_subscription!);
+        }
+    }
+
+    public void Dispose()
+    {
+        if (_resetTimer is not null)
+        {
+            // If we have a timer, we should reset debugging now
+            // otherwise we'll be permanently in debug mode
+            ResetDebugging();
+            _resetTimer.Dispose();
+        }
+
+        _discoveryService.RemoveSubscription(HandleConfigUpdate);
+
+        if (_subscription is { } subscription)
+        {
+            _subscriptionManager.Unsubscribe(subscription);
+        }
+    }
+
+    private static void ResetDebugging() => GlobalSettings.SetDebugEnabledInternal(false);
+
+    private async Task<ApplyDetails[]> RcmProductReceived(
+        Dictionary<string, List<RemoteConfiguration>> configByProduct,
+        Dictionary<string, List<RemoteConfigurationPath>>? removedConfigByProduct)
+    {
+        // We only expect _one_ of these to happen at a time, but handle the case where they all come together
+        ApplyDetails[]? results = null;
+        if (configByProduct.TryGetValue(RcmProducts.TracerFlareInitiated, out var initiatedConfig)
+         && initiatedConfig.Count > 0)
+        {
+            results = HandleTracerFlareInitiated(initiatedConfig);
+        }
+
+        if (configByProduct.TryGetValue(RcmProducts.TracerFlareRequested, out var requestedConfig)
+         && requestedConfig.Count > 0)
+        {
+            var handled = await HandleTracerFlareRequested(requestedConfig).ConfigureAwait(false);
+            results = results is null ? handled : [..results, ..handled];
+        }
+
+        if (removedConfigByProduct?.TryGetValue(RcmProducts.TracerFlareInitiated, out var removedConfig) == true
+         && removedConfig.Count > 0)
+        {
+            var handled = HandleTracerFlareResolved(removedConfig);
+            results = results is null ? handled : [..results, ..handled];
+        }
+
+        return results ?? [];
+    }
+
+    private ApplyDetails[] HandleTracerFlareInitiated(List<RemoteConfiguration> config)
+    {
+        try
+        {
+            // This product means "prepare for sending a tracer flare."
+            // We may consider doing more than just enabling debug mode in the future
+            GlobalSettings.SetDebugEnabledInternal(true);
+
+            // The timer is a fallback, in case we never receive a "send flare" product
+            var timer = new Timer(
+                _ => ResetDebugging(),
+                state: null,
+                dueTime: TimeSpan.FromMinutes(RevertGlobalDebugMinutes),
+                period: Timeout.InfiniteTimeSpan);
```

Comment:
Should we log a warning if we don't get the send flare after this timeout and disable debug mode?
I'm not entirely sure what the cases are where we wouldn't get a send flare are though.

Reply:
_Maaaaybe_? 🤔 It would require explicitly tracking whether we have sent _this_ flare (there could be multiple), and I think we would already be able to detect that from the existing logs, so I'm inclined to punt for now, and add it in later if it seems necessary if that's ok with you? (mostly to avoid another CI run tbh 😅)